provider/k8s.go

```diff
@@ -659,6 +661,149 @@ func (store genericFileStore) NumRows(key string) (int64, error) {
 	}
 }
 
+type csvIterator struct {
+	reader        *csv.Reader
+	currentValues GenericRecord
+	err           error
+	columnNames   []string
+	idx           int64
+	limit         int64
+}
+
+func (c *csvIterator) Next() bool {
+	if c.idx >= c.limit {
+		return false
+	}
+	row, err := c.reader.Read()
+	if err != nil {
+		if err == io.EOF {
+			return false
+		} else {
+			c.err = err
+			return false
+		}
+	}
+	c.currentValues = c.ParseRow(row)
+	c.idx += 1
+	return true
+}
+
+func (c *csvIterator) Values() GenericRecord {
+	return c.currentValues
+}
+
+func (c *csvIterator) Columns() []string {
+	return c.columnNames
+}
+
+func (c *csvIterator) Err() error {
+	return c.err
+}
+
+func (c *csvIterator) Close() error {
+	return nil
+}
+
+func (c *csvIterator) ParseRow(row []string) GenericRecord {
+	records := make(GenericRecord, len(row))
+	for i, value := range row {
+		if integer, err := strconv.Atoi(value); err == nil {
+			records[i] = integer
+			continue
+		}
+		if float, err := strconv.ParseFloat(value, 64); err == nil {
+			records[i] = float
+			continue
+		}
+		records[i] = value
+	}
+	return records
+}
+
+func newCSVIterator(b []byte, limit int64) (GenericTableIterator, error) {
```

Comment:
would these be used outside of the package?

Reply:
This should go in the filestore package whenever we actually get that out